
<launch>
  <arg name="rviz" default="false" />
  <arg name="robot_ns" default="spot" />

  <!-- Load base YAML files -->
  <rosparam file="$(find map_manager)/cfg/occupancy_map_param_ugv.yaml" ns="$(arg robot_ns)/occupancy_map" />
  <rosparam file="$(find map_manager)/cfg/object_map_param.yaml" ns="$(arg robot_ns)/occupancy_map" />
  <param name="use_sim_time" value="false" />


  <!-- Start mapping node -->
  <node pkg="map_manager" type="occupancy_map_node" name="occupancy_map" ns="$(arg robot_ns)">
    <!-- Override specific parameters -->
    <param name="range_sensor_input_mode" value="1" />
    <param name="sensor_fusion_mode" value="1" />
    <param name="point_cloud_topic" value="/ouster/points" />
    <param name="odom_topic" value="/spot/odom" />
    <rosparam param="body_to_range_sensor"> [1.0, 0.0, 0.0, 0.4, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.43, 0.0, 0.0, 0.0, 1.0]</rosparam>
	  <rosparam param="body_to_color_sensor"> [1.0, 0.0, 0.0, 0.4, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.13, 0.0, 0.0, 0.0, 1.0]</rosparam>
    <!-- detection -->
    <rosparam param="object_of_interest"> [0, 56, 57, 60]</rosparam>
    <rosparam param="object_bbox_color"> 
      [1.0, 0.0, 0.0,
       0.0, 1.0, 0.0,
       0.0, 0.0, 1.0,
       1.0, 1.0, 0.0]
    </rosparam>
    <rosparam param="object_class_label"> ["person", "chair", "couch", "dining table"]</rosparam>
    <param name="dbscan_min_points_cluster" value="30" />
    
    <!-- Occ map size -->
    <rosparam param="map_size_max">[9, 16, 3]</rosparam>
    <rosparam param="map_size_min">[-2, -20, -0.4]</rosparam>
    <!-- Local update range -->
    <rosparam param="local_update_range">[5, 5, 3]</rosparam>
    <!-- Visualization -->
    <rosparam param="local_map_size">[5, 5, 3]</rosparam>
    <param name="max_height_visualization" value="3.0" />
	  <param name="visualize_global_map" value="true" />
    <!-- Object map -->
    <param name="dbscan_downsample_factor" value="2" />
    <param name="dbscan_search_range_epsilon" value="0.08" />
    <param name="object_track_thresh" value="0.2" />
    <param name="object_map_debug" value="false" />
    <param name="debug_root_dir" value="/home/xiaoyang/spot_ws/debug_data/" />
  </node>

  <!-- RViz -->
  <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find map_manager)/rviz/spot_map_lio.rviz" />
  </group>
</launch>